ff/src/fields/models/quadratic_extension.rs

```diff
@@ -52,42 +52,38 @@ pub trait QuadExtConfig: 'static + Send + Sync + Sized {
     /// the quadratic non-residue. This is used in Karatsuba multiplication
     /// and in complex squaring.
     #[inline(always)]
-    fn mul_base_field_by_nonresidue(fe: &Self::BaseField) -> Self::BaseField {
-        Self::NONRESIDUE * fe
+    fn mul_base_field_by_nonresidue_in_place(fe: &mut Self::BaseField) -> &mut Self::BaseField {
+        *fe *= &Self::NONRESIDUE;
+        fe
     }
 
-    /// A specializable method for computing x + mul_base_field_by_nonresidue(y)
+    /// A specializable method for setting `y = x + NONRESIDUE * y`.
     /// This allows for optimizations when the non-residue is
     /// canonically negative in the field.
     #[inline(always)]
-    fn add_and_mul_base_field_by_nonresidue(
-        x: &Self::BaseField,
-        y: &Self::BaseField,
-    ) -> Self::BaseField {
-        *x + Self::mul_base_field_by_nonresidue(y)
+    fn add_and_mul_base_field_by_nonresidue_in_place(y: &mut Self::BaseField, x: &Self::BaseField) {
```

Comment:
question about the name: shouldn't it be "mul_base_field_by_nonresidue_and_add"?

Reply:
We can skip this issue for now and handle it in another PR.

Reply:
You're right. I'll do this renaming.